File: yurt-data/yurt-data-jdbc/src/main/java/com/github/liulus/yurt/jdbc/SQLExecutor.java
package com.github.liulus.yurt.jdbc;

import java.util.Collection;
import java.util.List;

/**
 * @author liulu
 * @version V1.0
 * @since 2020/11/10
 */
public interface SQLExecutor {

    <E> long insert(E entity);

    <E> int updateIgnoreNull(E entity);

    <E> int deleteById(Class<E> eClass, Long id);

    <E> E selectById(Class<E> eClass, Long id);

    <E> List<E> selectByIds(Class<E> eClass, Collection<Long> ids);

    <E> E selectForObject(SQL sql, Object params, Class<E> requiredType);

    <E> List<E> selectForList(SQL sql, Object params, Class<E> requiredType);

    <E> List<E> selectForPage(SQL sql, Object params, int pageNum, int pageSize, Class<E> requiredType);

    long count(SQL var1, Object params);

    int update(SQL var1, Object params);


}

File: yurt-data/yurt-data-jdbc/src/main/kotlin/com/github/liulus/yurt/jdbc/SQLExecutor.kt
package com.github.liulus.yurt.jdbc

/**
 *
 * @author liulu
 * @version V1.0
 * @since 2020/11/4
 */
interface SQLExecutor {

    fun <E> insert(entity: E): Long

    fun <E> updateIgnoreNull(entity: E): Int

    fun <E> deleteById(eClass: Class<E>, id: Long): Int

    fun <E> selectById(eCLass: Class<E>, id: Long): E?

    fun <E> selectByIds(eClass: Class<E>, ids: Collection<Long>): List<E>

    fun <E> selectForObject(sql: SQL, params: Any?, requiredType: Class<E>): E?

    fun <E> selectForList(sql: SQL, params: Any?, requiredType: Class<E>): List<E>

    fun count(sql: SQL, params: Any?): Long

    fun update(sql: SQL, params: Any?): Int
}